hr_holidays_public/models/resource_calendar.py

```diff
@@ -15,20 +15,46 @@ class ResourceCalendar(models.Model):
     def _attendance_intervals_batch_exclude_public_holidays(
         self, start_dt, end_dt, intervals, resources, tz
     ):
-        list_by_dates = (
-            self.env["hr.holidays.public"]
-            .get_holidays_list(
-                start_dt=start_dt.date(),
-                end_dt=end_dt.date(),
-                employee_id=self.env.context.get("employee_id", False),
+        holidays_by_country = {
+            False: set(
+                self.env["hr.holidays.public"]
+                .get_holidays_list(
+                    start_dt=start_dt.date(),
+                    end_dt=end_dt.date(),
+                    employee_id=self.env.context.get("employee_id", False),
+                )
+                .mapped("date")
             )
-            .mapped("date")
-        )
+        }
+        if resources:
+            employees = self.env["hr.employee.public"].search(
+                [("resource_id", "in", resources.ids)]
+            )
+            resource_country = {}
+            for employee in employees:
+                country = employee.address_id.country_id
+                resource_country[employee.resource_id.id] = country.id
+                if country.id not in holidays_by_country:
+                    holidays_by_country[country.id] = set(
```

Comment:
to be more clear about my PR https://github.com/gurneyalex/hr-holidays/pull/2
holidays will be assigned as a set of public holidays or empty set on line 55.